R/discrmd.R

```diff
@@ -87,22 +88,22 @@ drmd_psm <- function(ptdata, dpam, psmtype="simple", Ty=10, discrate=0, lifetabl
   ds <- ds |>
     dplyr::mutate(
       # Derive the background mortality for this timepoint
-      cqx = 1 - dplyr::lead(clx)/clx,
+      cqx = 1 - dplyr::lead(.data$clx)/.data$clx,
```

Comment:
What is your motivation for prefixing all the variables with `.data$`? Are you trying to remove `R CMD check` NOTEs about undefined global variables?

An alternative to prefixing with `.data$` is to add the variables you use in tidyverse functions to `utils::globalVariables()` (you'd also need to add `utils` to Imports). You can see how we do this in [{gsDesign2}](https://github.com/Merck/gsDesign2/blob/f20836472c1e341a73741129d70654faa3ec6a77/R/globals.R#L19) and [{simtrial}](https://github.com/Merck/simtrial/blob/fedf01d86f0e318a371ff50dac15156b1a2ffe9f/R/global.R#L25)

Reply:
That's probably a better way! Yes it's to remove R CMD check notes. There are various other functions with the same issue/behavior, so I should go through them all - thanks for showing me the better practice in gsDesign2 and simtrial.